sycl/test/format.py

```diff
@@ -0,0 +1,107 @@
+        # We traverse build/sycl/include/sycl directory
+        source_path = os.path.join(localConfig.sycl_include, "sycl")
+
+        # Optional filter can be passed through command line options
+        headers_filter = localConfig.sycl_headers_filter
+        for dirpath, _, filenames in os.walk(source_path):
+            relative_dirpath = dirpath[len(localConfig.sycl_include) + 1 :]
+            for filename in filenames:
+                if not filename.endswith(".hpp"):
+                    continue
+                filepath = os.path.join(dirpath, filename)
+
+                if headers_filter is not None:
+                    # Skip headers that doesn't match passed regexp
+                    if re.match(headers_filter, filepath) is None:
+                        continue
+                for t in self.getTestsForPath(
+                    testSuite,
+                    path_in_suite
+                    + (
+                        relative_dirpath,
+                        filename,
+                    ),
+                    filepath,
+                    litConfig,
+                    localConfig,
+                ):
+                    yield t
+
+    def execute(self, test, litConfig):
+        # List of files that are not expected to compile cleanly when used
+        # standalone. `os.path.join` is required here so the filtering works
+        # cross-platform
+        xfail = [
+            os.path.join(
+                "sycl", "ext", "oneapi", "experimental", "backend", "cuda.hpp"
+            ),
+            os.path.join(
+                "sycl", "ext", "intel", "esimd", "detail", "types_elementary.hpp"
+            ),
+        ]
+
+        command = [
+            test.config.clang,
+            "-fsycl",
+            "-fsyntax-only",
+            "-include",
+            test.file_path,
+            os.path.join(
+                test.suite.getSourcePath(("self-contained-headers",)),
+                "Inputs",
+                "test.cpp",
+            ),
+        ]
+
+        is_xfail = False
+        for path in xfail:
+            if test.file_path.endswith(path):
+                is_xfail = True
+                break
+
+        try:
+            out, err, exitCode = lit.util.executeCommand(
+                command,
+                # cwd=???,
+                env=test.config.environment,
+                timeout=litConfig.maxIndividualTestTime,
+            )
+            if is_xfail:
+                status = lit.Test.XPASS if exitCode == 0 else lit.Test.XFAIL
+            else:
+                status = lit.Test.PASS if exitCode == 0 else lit.Test.FAIL
+            timeoutInfo = None
+        except lit.util.ExecuteCommandTimeoutException as e:
+            out, err, exitCode, timeoutInfo = e.out, e.err, e.exitCode, e.msg
+            status = lit.Test.TIMEOUT
+
+        output = """Running command: %s\n""" % (" ".join(command),)
+        output += f"Exit Code: {exitCode}\n"
```

Comment:
There's a lot of `f`-strings mixed with `%` interpolation which is a bit confusing to read. I'd prefer they were all f-strings if possible

Reply:
Done in ac0ba9d78bf253ab2cfb8144be14470bab2a7dc7